File: bindings/php/MeshPhase.php
<?php
namespace Stencil;

final class MeshPhase {
    const BuildingAtlas = 0;
    const MeshingChunks = 1;
    const Complete = 2;
    const Failed = 3;

    public static function name(int $value): string {
        return match ($value) {
            self::BuildingAtlas => 'BuildingAtlas',
            self::MeshingChunks => 'MeshingChunks',
            self::Complete => 'Complete',
            self::Failed => 'Failed',
            default => "Unknown({$value})",
        };
    }
}

File: bindings/kotlin/src/main/kotlin/at/schem/nucleation/MeshPhase.kt
package at.schem.nucleation

import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure

internal interface MeshPhaseLib: Library {
}
/** Phase of a running [MeshJob].
*/
enum class MeshPhase {
    BuildingAtlas,
    MeshingChunks,
    Complete,
    Failed;

    fun toNative(): Int {
        return this.ordinal
    }


    companion object {
        internal val libClass: Class<MeshPhaseLib> = MeshPhaseLib::class.java
        internal val lib: MeshPhaseLib = Native.load("nucleation", libClass) 
        fun fromNative(native: Int): MeshPhase {
            return MeshPhase.entries[native]
        }

        fun default(): MeshPhase {
            return BuildingAtlas
        }
    }
}
